src/modules/hip/kernel/tensor_stddev.hpp

```diff
@@ -0,0 +1,894 @@
+
+    // Reduction of 16 floats on 16 threads per block in x dimension (for every y dimension)
+    for (int threadMax = 8; threadMax >= 1; threadMax /= 2)
+    {
+        if (hipThreadIdx_x < threadMax)
+        {
+            partialRVarRowPtr_smem[hipThreadIdx_x] += partialRVarRowPtr_smem[hipThreadIdx_x + threadMax];
+            partialGVarRowPtr_smem[hipThreadIdx_x] += partialGVarRowPtr_smem[hipThreadIdx_x + threadMax];
+            partialBVarRowPtr_smem[hipThreadIdx_x] += partialBVarRowPtr_smem[hipThreadIdx_x + threadMax];
+        }
+        __syncthreads();
+    }
+
+    if (hipThreadIdx_x == 0)
+    {
+        // Reduction of 16 floats on 16 threads per block in y dimension
+        for (int threadMax = 8, increment = 128; threadMax >= 1; threadMax /= 2, increment /= 2)
+        {
+            if (hipThreadIdx_y < threadMax)
+            {
+                partialRVarRowPtr_smem[0] += partialRVarRowPtr_smem[increment];
+                partialGVarRowPtr_smem[0] += partialGVarRowPtr_smem[increment];
+                partialBVarRowPtr_smem[0] += partialBVarRowPtr_smem[increment];
+            }
+            __syncthreads();
+        }
+
+        // Final store to dst
+        if (hipThreadIdx_y == 0)
+        {
+            int idx = ((hipBlockIdx_z * hipGridDim_y + hipBlockIdx_y) * hipGridDim_x + hipBlockIdx_x) * 3;
+            tensorVarArr[idx] = partialRVarRowPtr_smem[0];
+            tensorVarArr[idx + 1] = partialGVarRowPtr_smem[0];
+            tensorVarArr[idx + 2] = partialBVarRowPtr_smem[0];
+        }
+    }
+}
+
+// Compute complete tensor variance
+template <typename T, typename U>
+__global__ void tensor_var_pkd3_hip(T *srcPtr,
+                                    uint2 srcStridesNH,
+                                    U *tensorVarArr,
+                                    Rpp32f *mean,
+                                    RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    /* Stores individual channel Variations computed from Image Mean to compute Stddev of Image*/
+    __shared__ float partialChannelRVar_smem[16][16];
+    __shared__ float partialChannelGVar_smem[16][16];
+    __shared__ float partialChannelBVar_smem[16][16];
+    float *partialChannelRVarRowPtr_smem = &partialChannelRVar_smem[hipThreadIdx_y][0];
+    float *partialChannelGVarRowPtr_smem = &partialChannelGVar_smem[hipThreadIdx_y][0];
+    float *partialChannelBVarRowPtr_smem = &partialChannelBVar_smem[hipThreadIdx_y][0];
+    partialChannelRVarRowPtr_smem[hipThreadIdx_x] = 0.0f;
+    partialChannelGVarRowPtr_smem[hipThreadIdx_x] = 0.0f;
+    partialChannelBVarRowPtr_smem[hipThreadIdx_x] = 0.0f;
+
+    int index           = (id_z * 4) + 3;
+    float4 meanImage_f4 = (float4)mean[index];
```

Comment:
You could just say float4 meanImage_f4 = (float4)mean_f4[id_z].w; if you receive it as float4 *mean_f4 instead of Rpp32f *mean